include/utransport/UAttributes.h

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _UATTRIBUTES_
+#define _UATTRIBUTES_
+
+#include <memory>
+#include <include/uri/up_uri.h>
+#include <include/uuid/uuid_gen.h>
+#include "UMessageType.h"
+#include "USerializationHint.h"
+#include "UPriority.h"
+
+using namespace uri_datamodel;
+
+namespace uprotocol 
+{
+    namespace utransport
+    {
+        namespace datamodel
+        {
+            class UAttributes
+            {               
+                public:
+
+                    class UAttributesBuilder; 
+
+                    UAttributes() {
+                        this->id_ = nullptr;
+                        this->type_ = UMessageType::UNDEFINED;
+                        this->priority_ = UPriority::UNDEFINED;
+                        this->ttl_ = 0;
+                        this->token_ = "";
+                        this->hint_ = USerializationHint::UNKNOWN;
+                        this->sink_ = nullptr;
+                        this->plevel_ = 0;
+                        this->commstatus_ = 0;
```

Comment:
Should these two be enums ? What do these numbers mean ?

Reply:
they are defined as integers in the JAVA @tamarafischer  @stevenhartley  what do you think?